### DISCUSSION TITLE
Creating a static folder removes the default static files needed by the default template

### DISCUSSION THREAD

**Discussion by gerrod3** [rochacbruno/marmite#180] [General]
I wanted to change the default favicon that is generated so I added a favicon.ico in my source directory. Noticed it didn't change anything so I looked at the generated source and saw it was pulling it from `static/favicon.ico`, so I created a static folder and put it there. Then upon push to github and my webhoster building my site I notice none of the static files were loading. Looking at the logs I noticed none of the files were there, just the favicon I added, even though they were present in my local build. I think these lines getting the static files need to account for grabbing the default static files if the default template is being used:

https://github.com/rochacbruno/marmite/blob/0008518a8474b80db595d652a726d11d190ced74/src/site.rs#L691-L707

**Comment by rochacbruno** [comment]
The problem was here

https://github.com/rochacbruno/marmite/blob/8925fe060bb7979d9df7277ba88eb30e3bf7c034/src/site.rs#L669

It was copying favicon only to the root of the site, when it must copy to the static folder.

I already fixed as part of #174 

My idea is that you can drop a favicon.ico on the source folder (or content folder) and it will be copied to the output/static, no need to create your own static dir.

**Comment by gerrod3** [comment]
What about adding extra static files while using the default template? Seems like you have to manually copy the generated files if you want to add custom files.

**Comment by rochacbruno** [comment]
The current logic is that if static/ does not exist on the source folder, it will be copied from the embedded files to the output dir.

If it exists, then avoid copying so it will not override if using a custom theme.

You can override CSS and JS using the custom_ files, if more change is needed on the theme, then you can start your own theme.

```
marmite source --start-theme
```

Then it will copy templates and static from embedded to your source folder, and you can fully customize.

---

How can we improve?

Can you give me an example of a static that you would override or add? 

Maybe we add a config option like `merge_static: true` `--merge-static true`

**Comment by rochacbruno** [comment]
Resolved with file mapping feature
